src/client.zig

```diff
@@ -165,14 +154,13 @@ pub const Client = struct {
             .callback = callback,
             .operation = operation,
             .message = message.ref(),
-        }) catch {
-            callback(
+        }) catch |err| switch (err) {
+            .NoSpaceLeft => callback(
                 user_data,
                 operation,
-                empty_result[0..],
-                @enumToInt(ClientError.batch_not_queued),
-            );
-            return;
+                error.BatchNotQueued,
+            ),
```

Comment:
We still need the `return` after calling the callback, otherwise we would fall through outside of the switch and continue execution.

Reply:
thanks, good catch